### ISSUE TITLE
Process crash when both having and where contain subquery

### ISSUE THREAD

**Issue by JelteF** [citusdata/citus#3090]
Code to reproduce:
```sql
CREATE TABLE stock (
  s_w_id int NOT NULL,
  s_i_id int NOT NULL,
  s_order_cnt int NOT NULL
);

SELECT create_distributed_table('stock','s_w_id');

select     s_i_id, sum(s_order_cnt) as ordercount
from     stock
where   s_order_cnt > (select sum(s_order_cnt) * .005 from stock)
group by s_i_id
having   sum(s_order_cnt) > (select sum(s_order_cnt) * .005 from stock)
order by ordercount desc;
```

Probable cause: https://github.com/citusdata/citus/blob/018ad1c58e9c38af43ce49fc7b7cf4af1a36168b/src/backend/distributed/planner/multi_logical_planner.c#L832

That piece of code tries to check for subqueries in having by checking if there is no subquery in where. In this case there's a subquery in where, but also in having. So the check fails here and we pass validation even if we shouldn't, because we don't support subqueries in having (yet).

**Comment by onderkalaci** [comment]
Looks the same with #1882

**Comment by marcocitus** [comment]
Fixed by #3098
